src/widgets/index/index.ts

```diff
@@ -459,11 +459,16 @@ See https://www.algolia.com/doc/guides/building-search-ui/widgets/customize-an-e
         }
       });
 
-      derivedHelper.on('result', () => {
+      derivedHelper.on('result', ({ results }) => {
         // The index does not render the results it schedules a new render
         // to let all the other indices emit their own results. It allows us to
         // run the render process in one pass.
         instantSearchInstance.scheduleRender();
+
+        // the derived helper is the one which actually searches, but the 'main'
+        // helper is exposed e.g. via instance.helper, which needs to have the
+        // results accessible as well.
```

Comment:
The comment is misleading. We expose the [`mainIndex.helper`](https://github.com/algolia/instantsearch.js/blob/57cd9d7c4806239a36cc7886177c2698bf9052a0/src/lib/InstantSearch.ts#L430-L432) via `instantSearchInstance.helper`  not the `mainHelper`. The `mainHelper` is the one used to trigger the requests, it is not tied to any states nor results.

Reply:
thanks, that's a good point. Reworded it in be8885a